noir/noir-repo/acvm-repo/acir_field/src/field_element.rs

```diff
@@ -139,17 +139,25 @@ impl<'de, T: ark_ff::PrimeField> Deserialize<'de> for FieldElement<T> {
 
 impl<F: PrimeField> From<u128> for FieldElement<F> {
     fn from(a: u128) -> FieldElement<F> {
-        let result = match F::from_str(&a.to_string()) {
-            Ok(result) => result,
-            Err(_) => panic!("Cannot convert u128 as a string to a field element"),
-        };
-        FieldElement(result)
+        FieldElement(F::from(a))
     }
 }
 
 impl<F: PrimeField> From<usize> for FieldElement<F> {
     fn from(a: usize) -> FieldElement<F> {
-        FieldElement::from(a as u128)
+        FieldElement(F::from(a as u64))
```

Comment:
```suggestion
        FieldElement::from(a as u64)
```
Nit: We made a `From<u64> for FieldElement<F>` below should be able to use that and just change the cast.